internal/cli/command/secret/create.go

```diff
@@ -373,14 +373,23 @@ func surveySecretFields(options *createSecretOptions, secretTypes map[string]pip
 		fields := secretType.Fields
 		questions := make([]secretFieldQuestion, len(fields))
 		for index, f := range fields {
-			questions[index] = secretFieldQuestion{
-				name: f.Name,
-				input: &survey.Password{
+			sfq := secretFieldQuestion{
+				name:     f.Name,
+				required: f.Required,
+			}
+
+			if f.IsPassword != nil && *f.IsPassword {
```

Comment:
**Required**: as we discussed in private, this condition is going to need an update one way or another.